### DISCUSSION TITLE
Curiosity: radio buttons vs checkboxes

### DISCUSSION THREAD

**Discussion by mergen3107** [koreader/koreader#8167] [Q&A]
This is just a question out of my curiosity. In KOReader, why checkboxes are used in places where radio buttons are expected / more appropriate? I mean the aesthetics. 

AFAIK, in general, radio buttons are for items that are mutually exclusive, thus you can select only one item out of the list. Checkboxes are for lists where multiples items can be selected. 

For example, in choosing Fonts, there is only one font that can be chosen at once. Or in Highlight drawing style (Underline, Lighten, Invert). Or in many other places (Stable vs Development update channel, etc). 

Radio buttons are only used in keyboard layout dialog and selecting render engine for a book, CRengine vs MuPDF, (or maybe in some other places I don’t know yet like plugins I don’t use). 
Is it related to some legacy complications?

**Comment by Frenzie** [comment]
One complication is that I consider radiobuttons overrated things primarily understood by designers and programmers so I never bothered writing any at first. Cf. <https://uxmovement.com/forms/why-radio-buttons-and-checkboxes-cant-co-exist/>. Besides which, the font menu was literally about the only place where it would've been useful at the time, and it was already using checkmarks going back years more. You'll note there was about a year in between when I wrote the two.

I'm not convinced that clarity would be helped by replacing the font menu with radio buttons, but feel free to switch it. Looks a little something like this (but you'd have to do a bit more than just swapping out checkmark for radiobutton like I did to make this screenshot).

![Screenshot_2021-09-01_17-43-12](https://user-images.githubusercontent.com/202757/131702078-da5d3c9e-bcd1-4e9e-a17a-323605ee228d.png)

**Reply by mergen3107** [reply]
Thanks for the insight!
Yeah, I see the point. In Top menu there are sometimes line-separated groups of settings with single/multiple choices, so it would look inconsistent if radio buttons are mixed with check boxes. 

So would you rather have only checkboxes everywhere?

**Reply by Frenzie** [reply]
I have seen no indications that radiobuttons are actively harmful or anything, and they do provide some extra contextual information to people in the know. The behavior *is* different after all. As such I'm not convinced that the consistency principle as in the article I linked necessarily applies. I've just noticed in UX experiments that many people don't know the difference, and if you take that as a given then logically you're just distracting people by presenting a difference. Some will quite literally remark that a checkmark would be clearer and I can't help but agree that it's clearer to see if something is in fact checked. Also in menus specifically I think radiobuttons look a bit ugly but that shouldn't be the main consideration. :-)

**Reply by mergen3107** [reply]
I see :) Thank you for your input! It is always interesting to see the "kitchen" of the UI/UX and decisions behind them.
